### PR TITLE
[2778] Authorize empty string for direct edit

### LINKED ISSUE #2778: Leaving direct edit with empty text doesn't close the input
* [*] **I have specified the version** :
- latest (2023.12.0)

### Actual behavior

direct_edit_empty

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/2778

# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by AxelRICHARD — APPROVED** [review]

**Review by sbegaudeau — DISMISSED** [review]
I don't have any issue with the code, it looks exactly as it should but could you update the test suite `project/diagrams/graphical-dnd.cy.js` to our new standards and fix its flakiness.

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="24.036" tests="1" failures="1">
  <testsuite name="Root Suite" timestamp="2023-12-08T16:36:07" tests="0" file="cypress/e2e/project/diagrams/graphical-dnd.cy.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="/projects/:projectId/edit - graphical DnD" timestamp="2023-12-08T16:36:07" tests="1" time="24.022" failures="1">
    <testcase name="/projects/:projectId/edit - graphical DnD Check graphical Drag and Drop" time="0.000" classname="Check graphical Drag and Drop">
      <failure message="expected false to be true" type="AssertionError"><![CDATA[AssertionError: expected false to be true
    at Context.eval (webpack:///./cypress/support/reactFlowCommands.js:51:64)

      + expected - actual:  failed to generate Mocha diff
]]></failure>
    </testcase>
  </testsuite>
</testsuites>
```

Once you will have updated the pull request, I won't require you to update yet another test. Have a look at the explorer.cy.ts suite to see how you could improve it and don't hesitate to talk to me if you have any question. An improvement to our integration tests to help validate your contribution would be nice too.

I'm sorry to have to force everybody to help in this effort of improving our integration tests but it won't move forward otherwise.

**Review by pcdavid — APPROVED** [review]
This should probably also apply to `TreeItemDirectEditInput`?

File: integration-tests/cypress/workbench/Details.ts [hunk 1]
```diff
@@ -19,4 +19,16 @@ export class Details {
   public getTextField(label: string): Cypress.Chainable<JQuery<HTMLInputElement | HTMLTextAreaElement>> {
     return this.getDetailsView().find(`[data-testid="input-${label}"]`);
   }
+
+  public getCheckBox(label: string): Cypress.Chainable<JQuery<HTMLInputElement>> {
+    return this.getDetailsView().findByTestId(`${label}`).find('input');
```
Comment (pcdavid) [inline integration-tests/cypress/workbench/Details.ts]: Just `label`?

File: integration-tests/cypress/workbench/Details.ts [hunk 2]
```diff
@@ -19,4 +19,16 @@ export class Details {
...
+  }
+
+  public openReferenceWidgetOptions(label: string): void {
+    this.getDetailsView().findByTestId(`${label}`).find(`button[title="Open"]`).click();
```
Comment (pcdavid) [inline integration-tests/cypress/workbench/Details.ts]: Just `label`?

Comment (pcdavid) [reply integration-tests/cypress/workbench/Details.ts]: No \` needed for the `'buton[title="Open"]'` string.

**Review by pcdavid — APPROVED** [review]
Only minor remarks.

File: integration-tests/cypress/workbench/Details.ts (hunk 1 above)
Comment (frouene) [reply integration-tests/cypress/workbench/Details.ts]: done

File: integration-tests/cypress/workbench/Details.ts (hunk 2 above)
Comment (frouene) [reply integration-tests/cypress/workbench/Details.ts]: done

**Pull request merged**